### PR TITLE
fix: extract more Codex Cursor and Gemini screen metadata

### PR DESCRIPTION
**Pull request by EtanHey**
## Summary
- broaden Codex header parsing so model extraction still works when  is absent
- parse Cursor token counts from  status lines and accept bare  context strips
- support Gemini  lines and  status markers with regression tests

## Test plan
- bun run test tests/screen-parser.test.ts
- bun run test
- bun run typecheck
- bun run build
- real MCP stdio client verification against  on live cmux surfaces

## Notes
- CodeRabbit pre-commit review could not run because the review service was rate-limited during this session

---

> [!NOTE]
> **Medium Risk**
> Regex changes affect agent-type detection and status/model extraction, which could cause misclassification on unseen screen formats. Risk is mitigated by added targeted regression tests for the new patterns.
> 
> **Overview**
> Improves `parseScreen` metadata extraction across multiple CLIs by broadening Codex header matching to extract the model even when the `% left` segment is missing, and by parsing context-left percent separately (including trailing whitespace).
> 
> Extends Cursor parsing to read token counts from activity lines (e.g. “Calling … tokens”) and relaxes context-strip parsing so `Auto · NN%` works without a trailing separator.
> 
> Enhances Gemini parsing to accept bullet-prefixed `Model:` lines and to treat standalone `Working`/`✦ Working` status lines as `working` without misclassifying prose, with new tests covering these cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 7cba78c18376b0ceab2d05bf03cbacd2308378a7. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

> [!NOTE]
> ### Fix screen metadata extraction for Codex, Cursor, and Gemini agents
> Improves [screen-parser.ts](https://github.com/EtanHey/cmuxlayer/pull/73/files#diff-10170f5d46ab5ce45acb8040a69928bafe44d7f458eb4af90a211557ae574e50) to handle more header and status line variants across three agent types:
>
> - Codex: new `CODEX_HEADER_RE` matches headers with or without context-left and trailing segments; `CODEX_CONTEXT_LEFT_RE` handles `% left` parsing with extra trailing whitespace
> - Gemini: `GEMINI_MODEL_RE` now accepts bullet-prefixed `- Model:` lines; `GEMINI_WORKING_RE` detects `✦ Working` status lines; agent detection no longer misclassifies Codex headers as Gemini
> - Cursor: `CURSOR_ACTIVITY_TOKEN_RE` parses token counts from activity lines like `Calling 1.59k tokens`; `CURSOR_STATUS_PCT_RE` now tolerates missing trailing middle-dot segment
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 7cba78c.</sup>
> 
>

## Release Notes

* **Bug Fixes**
  * Improved detection and parsing accuracy for Codex, Gemini, and Cursor models across various output formats
  * Enhanced status and token count information extraction with more robust fallback mechanisms
  * Better handling of edge cases in model identification and percent-status parsing

* **Tests**
  * Expanded test coverage for model detection, status parsing, and token count extraction scenarios

### CODE REVIEW

**Comment by EtanHey** [comment]
@coderabbitai review

**Comment by EtanHey** [comment]
@codex review

**Comment by EtanHey** [comment]
@cursor @bugbot review

**Comment by EtanHey** [comment]
Addressed the unresolved parser review findings. @coderabbitai review

@codex review

@cursor @bugbot re-review

File: src/screen-parser.ts [hunk 1]
```diff
@@ -93,14 +93,15 @@ const MODEL_EMOJI_RE =
...
+const CODEX_HEADER_RE =
+  /^(gpt-[0-9][0-9a-z.-]*(?:\s+\w+)?)\s*(?:[·•]\s*(?:(\d+)%\s+left(?:\s*[·•]\s*[^\n]+)?|[^\n]+))$/m;
 const CODEX_WORKING_RE =
   /Working\s*\(([0-9]+m\s*[0-9]+s)\s*[•·]\s*esc to interrupt\)/i;
 const CODEX_RESUME_RE = /To continue this session,\s*run\s+codex\s+resume/i;
 const CODEX_ACTION_RE = /^\s*[•·]\s+(.+)$/gm;
 const GEMINI_MODEL_RE =
-  /(?:^|\n)\s*(?:Model:\s*)?(gemini-[0-9][0-9a-z.-]*)\b/im;
+  /(?:^|\n)\s*(?:-\s*)?(?:Model:\s*)?(gemini-[0-9][0-9a-z.-]*)\b/im;
+const GEMINI_WORKING_RE = /(?:^|\n)\s*(?:✦\s*)?Working\b/im;
```
[the remark this answers is not in the document]
Comment (EtanHey) [reply src/screen-parser.ts]: Fixed in `7cba78c`. `GEMINI_WORKING_RE` is now restricted to standalone status lines, and `tests/screen-parser.test.ts` includes a regression case for `Working with existing APIs...` staying `idle`.

File: src/screen-parser.ts [hunk 2]
```diff
@@ -93,14 +93,15 @@ const MODEL_EMOJI_RE =
 // Last resort: 🤖 + bare model family name (for narrow panes where version is cut off)
 const MODEL_KEYWORD_RE = /🤖\s*(Opus|Sonnet|Haiku)\b/i;
 const EXIT_CODE_RE = /(?:exit(?:ed)?\s+with\s+code|code)\s+(\d+)/gi;
-const CODEX_MODEL_RE =
-  /^(gpt-[0-9][0-9a-z.-]*(?:\s+\w+)?)\s*[·•]\s*(\d+)%\s+left/m;
+const CODEX_HEADER_RE =
+  /^(gpt-[0-9][0-9a-z.-]*(?:\s+\w+)?)\s*(?:[·•]\s*(?:(\d+)%\s+left(?:\s*[·•]\s*[^\n]+)?|[^\n]+))$/m;
```
[the remark this answers is not in the document]
Comment (EtanHey) [reply src/screen-parser.ts]: Fixed in `7cba78c`. I split Codex context capture into a dedicated `CODEX_CONTEXT_LEFT_RE` and added a regression test covering trailing whitespace after `% left`.
